xarray/testing.py

```diff
@@ -0,0 +1,143 @@
+"""Testing functions exposed to the user API"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from xarray.core import ops
+
+
+def _decode_string_data(data):
+    if data.dtype.kind == 'S':
+        return np.core.defchararray.decode(data, 'utf-8', 'replace')
+    return data
+
+
+def _data_allclose_or_equiv(arr1, arr2, rtol=1e-05, atol=1e-08,
+                            decode_bytes=True):
+    if any(arr.dtype.kind == 'S' for arr in [arr1, arr2]) and decode_bytes:
+        arr1 = _decode_string_data(arr1)
+        arr2 = _decode_string_data(arr2)
+    exact_dtypes = ['M', 'm', 'O', 'U']
+    if any(arr.dtype.kind in exact_dtypes for arr in [arr1, arr2]):
+        return ops.array_equiv(arr1, arr2)
+    else:
+        return ops.allclose_or_equiv(arr1, arr2, rtol=rtol, atol=atol)
+
+
+def assert_equal(a, b):
+    """Like :py:func:`numpy.testing.assert_array_equal`, but for xarray
+    objects.
+
+    Raises an AssertionError if two objects are not equal. This will match
+    data values and coordinates, but not names or attributes (except for
+    Dataset objects for which the variable names must match).
+
+    Parameters
+    ----------
+    a : xarray.Dataset or xarray.DataArray
+        The first object to compare
+    b : xarray.Dataset or xarray.DataArray
+        The second object to compare
+
+    See also
+    --------
+    assert_identical, assert_allclose, xarray.Dataset.equals,
```

Comment:
I think "See also" needs items separated by new lines, not commas. But I haven't checked what sphinx actually does.

Also: I would add `numpy.testing.assert_array_equal` here.